diff --git a/src/include/ndpi_private.h b/src/include/ndpi_private.h
index adce06815..7a067e7b9 100644
--- a/src/include/ndpi_private.h
+++ b/src/include/ndpi_private.h
@@ -1007,6 +1007,7 @@ void init_paltalk_dissector(struct ndpi_detection_module_struct *ndpi_struct, u_
 void init_dicom_dissector(struct ndpi_detection_module_struct *ndpi_struct, u_int32_t *id);
 void init_lagofast_dissector(struct ndpi_detection_module_struct *ndpi_struct, u_int32_t *id);
 void init_gearup_booster_dissector(struct ndpi_detection_module_struct *ndpi_struct, u_int32_t *id);
+void init_xml_rpc_dissector(struct ndpi_detection_module_struct *ndpi_struct, u_int32_t *id);
 
 
 #ifdef CUSTOM_NDPI_PROTOCOLS
diff --git a/src/include/ndpi_protocol_ids.h b/src/include/ndpi_protocol_ids.h
index 54fb38ffb..192578c1b 100644
--- a/src/include/ndpi_protocol_ids.h
+++ b/src/include/ndpi_protocol_ids.h
@@ -474,6 +474,7 @@ typedef enum {
   NDPI_PROTOCOL_RUTUBE                = 443,
   NDPI_PROTOCOL_LAGOFAST              = 444,
   NDPI_PROTOCOL_GEARUP_BOOSTER        = 445,
+  NDPI_PROTOCOL_XML_RPC              = 446,
 
 #ifdef CUSTOM_NDPI_PROTOCOLS
 #include "../../../nDPI-custom/custom_ndpi_protocol_ids.h"
diff --git a/src/lib/ndpi_main.c b/src/lib/ndpi_main.c
index 0aa7d5646..7fe03505f 100644
--- a/src/lib/ndpi_main.c
+++ b/src/lib/ndpi_main.c
@@ -2481,6 +2481,12 @@ static void ndpi_init_protocol_defaults(struct ndpi_detection_module_struct *ndp
                           ndpi_build_default_ports(ports_a, 0, 0, 0, 0, 0) /* TCP */,
                           ndpi_build_default_ports(ports_b, 0, 0, 0, 0, 0) /* UDP */);
 
+  ndpi_set_proto_defaults(ndpi_str, 1 /* cleartext */, 0 /* nw proto */, NDPI_PROTOCOL_ACCEPTABLE, NDPI_PROTOCOL_XML_RPC,
+                          "XML-RPC", NDPI_PROTOCOL_CATEGORY_RPC,
+                          ndpi_build_default_ports(ports_a, 0, 0, 0, 0, 0) /* TCP */,
+                          ndpi_build_default_ports(ports_b, 0, 0, 0, 0, 0) /* UDP */);
+
+
 #ifdef CUSTOM_NDPI_PROTOCOLS
 #include "../../../nDPI-custom/custom_ndpi_main.c"
 #endif
@@ -6390,6 +6396,10 @@ static int ndpi_callback_init(struct ndpi_detection_module_struct *ndpi_str) {
   /* GearUP Booster */
   init_gearup_booster_dissector(ndpi_str, &a);
 
+  /* XML-RPC */
+  init_xml_rpc_dissector(ndpi_str, &a);
+
+
 #ifdef CUSTOM_NDPI_PROTOCOLS
 #include "../../../nDPI-custom/custom_ndpi_main_init.c"
 #endif
